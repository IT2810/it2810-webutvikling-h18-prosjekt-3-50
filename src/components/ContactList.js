import React, { Component } from 'react'
import { StyleSheet } from 'react-native'
import { List, ListItem, Text, Left, Right, Button, View, Row, Picker, Icon, Card } from 'native-base'
import { connect } from 'react-redux'

import { removeContact } from '../../actions/index'
const savedContacts = require('../assets/contacts.json')

export class ContactList extends Component {

  constructor(props, context) {
    super(props, context)
  }

  render () {
    return (
      <View>
        <Row style={{marginTop: 16}}>
          <Left>
            <Text style={styles.inputTitle}>Contacts</Text>
          </Left>
          <Right>
            <Button
              primary
              block
              onPress={() => this.props.navigation.navigate('AddContact')}
            >
              <Text>ADD CONTACT</Text>
            </Button>
          </Right>
        </Row>
        <Card style={styles.cardWithList}>
          {this.props.contacts.map((contact, index) => (
            <ListItem key={index}>
              <Left>
                <Text> { contact.name } </Text>
              </Left>
              <Right>
                <Button
                  danger
                  onPress={() => this.props.removeContact(contact)}
                >
                  <Icon active name="trash" />
                </Button>
              </Right>
            </ListItem>
          ))}
        </Card>
      </View>
    )
  }
}

function mapStateToProps(state){
  if (state.sessions.activeSession) {
    return {
      contacts: state.sessions.activeSession.contacts    }  
  } else {
    return {
      contacts: []
    }
  }
}

const mapDispatchToProps = dispatch => ({
  removeContact: contact => dispatch(removeContact(contact))
})

export default connect(mapStateToProps, mapDispatchToProps)(ContactList)

const styles = StyleSheet.create ({
  cardWithList: {
    marginTop: '5%',
    marginBottom: '5%'
  },
  inputTitle: {
    lineHeight: 45,
    fontSize: 27,
    marginLeft: '8%',
    fontWeight: '600'
  }
})
