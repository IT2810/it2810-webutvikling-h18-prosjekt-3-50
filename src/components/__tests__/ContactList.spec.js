/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ContactList from '../ContactList'
import { findByID } from '../../testUtils.js'

describe('ContactList', () => {
  let wrapper
  let navigation
  let dispatchRemoveContactMock
  let contacts

  beforeEach(() => {
    dispatchRemoveContactMock = jest.fn()
    navigateMock = jest.fn()
    navigation = { navigate: navigateMock }
    contacts = [{name: 'Test Name'}]
    wrapper = shallow(<ContactList
      navigation={navigation}
      removeContact={dispatchRemoveContactMock}
      contacts={contacts} />)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

})
