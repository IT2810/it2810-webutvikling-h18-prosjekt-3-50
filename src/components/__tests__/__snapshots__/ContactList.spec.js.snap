// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList renders correctly 1`] = `
<ViewNB
  style={Object {}}
>
  <RowNB
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <Styled(Left)>
      <Styled(Text)
        style={
          Object {
            "fontSize": 27,
            "fontWeight": "600",
            "lineHeight": 45,
            "marginLeft": "8%",
          }
        }
      >
        Contacts
      </Styled(Text)>
    </Styled(Left)>
    <Styled(Right)>
      <Styled(Button)
        block={true}
        onPress={[Function]}
        primary={true}
      >
        <Styled(Text)>
          ADD CONTACT
        </Styled(Text)>
      </Styled(Button)>
    </Styled(Right)>
  </RowNB>
  <Styled(Card)
    style={
      Object {
        "marginBottom": "5%",
        "marginTop": "5%",
      }
    }
  >
    <Styled(ListItem)
      key="0"
    >
      <Styled(Left)>
        <Styled(Text)>
           
          Ola Nordmann
           
        </Styled(Text)>
      </Styled(Left)>
      <Styled(Right)>
        <Styled(Button)
          danger={true}
          onPress={[Function]}
        >
          <Styled(Icon)
            active={true}
            name="trash"
          />
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      key="1"
    >
      <Styled(Left)>
        <Styled(Text)>
           
          Kari Nordmann
           
        </Styled(Text)>
      </Styled(Left)>
      <Styled(Right)>
        <Styled(Button)
          danger={true}
          onPress={[Function]}
        >
          <Styled(Icon)
            active={true}
            name="trash"
          />
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
  </Styled(Card)>
</ViewNB>
`;
